### PR TITLE
Fix runtime error on exception reraise

### PR DESCRIPTION
**Pull request by Newlifer**
Fix issue when we doesn't have access to write and try to reraise no active exception, causes runtime error without any info.

### CODE REVIEW

File: conans/util/files.py [hunk 1]
```diff
@@ -194,7 +194,7 @@ def _change_permissions(func, path, exc_info):
         os.chmod(path, stat.S_IWUSR)
         func(path)
     else:
-        raise
+        raise Exception("Cannot change permissions for {}!".format(path))
```
Comment (lasote) [inline conans/util/files.py]: Should we use `exc_info` to include some more information about the error?

**Comment by memsharded** [comment]
I am trying to add some tests for this contribution, but I cannot fire that behavior. Could you please @Newlifer clarify how could this be reproduced? Once you are inside the ``def _change_permissions(func, path, exc_info):`` it seems there is always an active exception, and its information is in the ``exc_info`` variable.

**Comment by memsharded** [comment]
Hi!
The tests are still broken. Could you please define a reproducible case that I could convert to a test for this change? Thanks!

**Comment by lasote** [comment]
I will merge it and fix it in develop to keep your contribution. Thanks!

**Pull request merged**